src/Middleware/CorsMiddlewareFactory.php

```diff
@@ -9,14 +9,22 @@
 use Mezzio\Cors\Service\ResponseFactoryInterface;
 use Psr\Container\ContainerInterface;
 
+use function assert;
+
 final class CorsMiddlewareFactory
 {
     public function __invoke(ContainerInterface $container): CorsMiddleware
     {
+        $cors = $container->get(CorsInterface::class);
+        assert($cors instanceof CorsInterface);
```

Comment:
@Ocramius is there something to avoid having this annotation stuff. I could use `ReflectionBasedFactory` but that would require the servicemanager dependency. Heading forward to the `container-contracts`, is that still something we are planning @weierophinney ?

Reply:
Could you rephrase the question? Nothing wrong with asserting on the type when it is `mixed`

Reply:
Oh, I meant "assertion stuff".
Well, I'd prefer having a psalm plugin (I did some more research and found [lctrs/psalm-psr-container-plugin](https://packagist.org/packages/lctrs/psalm-psr-container-plugin)) which makes these `assert` statements unneeded.

Just wondering if the usage of such a plugin is okay as the project itself provides a configuration for all that interface aliasing and imho its not needed to make assertions on all interfaces fetched from the container.

Could be quite heavy, depending on how many factories are in existing components for example.

Reply:
I don't think a plugin should be used it: there's no guarantee (besides our "best guess") that we'll get what we asked for.

Remember that the container may be any implementation, not just the one we wired up.

Reply:
Okay. So should we use `assert()` or `Assert` from `webmozart/assert` which is already a dependency in this package.

Reply:
I think for consistency with overall usage in the package, `webmozart/assert` can be used

Reply:
Remember that `assert()`, while more flexible, can be disabled via ini settings. If we already have the `webmozart/assert` dependency, let's use that one, which at least always throws 👍

Reply:
At best, we'll have `get()` return `object` or `array|object` in future revisions. Assertions are how we validate we got what we expected. The nice bit about using `assert()` for these is that you can set `zend.assertions = -1` in your `php.ini` in production and they'll be skipped by the engine, meaning no performance impact.

It seems like a PITA at first, and then you start recognizing all the subtle assumptions you have about the code, and how many of them can turn out wrong, and the bad things that snowball when they do. It's definitely worth the effort.